src/prelude.rs

```diff
@@ -0,0 +1,12 @@
+//! Crate-local import prelude, primarily intended as a facade for accessing
+//! heap-allocated data structures.
+
+#[cfg(all(feature = "alloc", not(feature = "std")))]
+pub use alloc::boxed::Box;
+#[cfg(all(feature = "alloc", not(feature = "std")))]
+pub use alloc::string::String;
+#[cfg(all(feature = "alloc", not(feature = "std")))]
+pub use alloc::vec::Vec;
+
+#[cfg(feature = "std")]
+pub use std::prelude::v1::*;
```

Comment:
This looks a little bit weird because we pull these types in from different places depending on the precise set of features enabled, so it doesn't look purely additive, however `std` just re-exports the `alloc` types as a facade, so it winds up being purely additive.